skyportal/facility_apis/lt.py

```diff
@@ -0,0 +1,855 @@
+        # Send cancel_payload, and receive response string, removing the encoding tag which causes issue with lxml parsing
+        response = client.service.handle_rtml(cancel).replace(
+            'encoding="ISO-8859-1"', ''
+        )
+        response_rtml = etree.fromstring(response)
+        mode = response_rtml.get('mode')
+        uid = response_rtml.get('uid')
+        if mode == 'confirm':
+            request.status = "deleted"
+        else:
+            request.status = f"failed to delete: {mode}"
+
+        transaction = FacilityTransaction(
+            request=http.serialize_requests_request_xml(cancel),
+            response=http.serialize_requests_response_xml(response),
+            followup_request=request,
+            initiator_id=request.last_modified_by_id,
+        )
+        DBSession().add(transaction)
+
+
+class IOOAPI(LTAPI):
+
+    """An interface to LT IOO operations."""
+
+    # subclasses *must* implement the method below
+    @staticmethod
+    def submit(request):
+
+        """Submit a follow-up request to LT's IOO.
+
+        Parameters
+        ----------
+        request: skyportal.models.FollowupRequest
+            The request to add to the queue and the SkyPortal database.
+        """
+
+        from ..models import DBSession, FacilityTransaction
+
+        altdata = request.allocation.load_altdata()
+        if not altdata:
+            return
+
+        ltreq = IOORequest()
+        observation_payload = ltreq._build_prolog()
+        ltreq._build_project(observation_payload, request)
+        ltreq._build_inst_schedule(observation_payload, request)
+
+        headers = {
+            'Username': altdata["username"],
+            'Password': altdata["password"],
+        }
+        url = '{0}://{1}:{2}/node_agent2/node_agent?wsdl'.format(
+            'http', cfg['app.lt_host'], cfg['app.lt_port']
+        )
+        client = Client(url=url, headers=headers)
+        full_payload = etree.tostring(
+            observation_payload, encoding="unicode", pretty_print=True
+        )
+        # Send payload, and receive response string, removing the encoding tag which causes issue with lxml parsing
+        response = client.service.handle_rtml(full_payload).replace(
+            'encoding="ISO-8859-1"', ''
+        )
+        response_rtml = etree.fromstring(response)
+        mode = response_rtml.get('mode')
+
+        if mode == 'confirm':
+            request.status = 'submitted'
+        else:
+            request.status = f'rejected: {mode}'
+
+        transaction = FacilityTransaction(
+            request=http.serialize_requests_request_xml(full_payload),
+            response=http.serialize_requests_response_xml(response),
+            followup_request=request,
+            initiator_id=request.last_modified_by_id,
+        )
+
+        DBSession().add(transaction)
+
+    _observation_types = ['r', 'gr', 'gri', 'griz', 'ugriz']
+    _exposure_types = ['1x120s', '2x150s']
```

Comment:
Make 2 fields, an integer field for the number of exposures and a floating point text field for the exposure time